objectserver/auditor.go

```diff
@@ -62,6 +67,77 @@ type Auditor struct {
 	bytesProcessed, totalBytes    int64
 	quarantines, totalQuarantines int64
 	errors, totalErrors           int64
+	ecfuncs                       ECAuditFuncs
+}
+
+type ECAuditFuncs interface {
+	AuditShard(path string, hash string, skipMd5 bool) (int64, error)
+	AuditNurseryObject(path string, metabytes []byte, skipMd5 bool) (int64, error)
+}
+
+type RealECAuditFuncs struct{}
+
+// AuditNurseryObject of indexdb shard, does nothing
+func (RealECAuditFuncs) AuditNurseryObject(path string, metabytes []byte, skipMd5 bool) (int64, error) {
+	finfo, err := os.Stat(path)
+	if err != nil || !finfo.Mode().IsRegular() {
+		// We're not going to do any quarantining here. It's likely the object
+		// simply got stabilized and is gone.
+		return 0, nil
+	}
+
+	bytes := int64(0)
+	if !skipMd5 {
+		metadata := map[string]string{}
+		if err = json.Unmarshal(metabytes, &metadata); err != nil {
+			return bytes, fmt.Errorf("Error decoding metadata: %s", err)
+		}
+		hash, ok := metadata["ETag"]
+		if !ok {
+			return bytes, fmt.Errorf("Metadata missing ETag: %s", string(metabytes))
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			return bytes, fmt.Errorf("Error opening file: %s", err)
+		}
+		defer file.Close()
+		h := md5.New()
+		bytes, err = common.Copy(file, h)
```

Comment:
should be rate limit this? seems like we could read chunks and sleep a bit between. for a 5GB file would just blow through it here and then sleep a long time below?